src/sqlfluff/rules/L046.py

```diff
@@ -88,10 +90,19 @@ def _eval(self, context: RuleContext) -> LintResult:
 
                 # Check the initial whitespace.
                 if not ws_pre or (ws_pre != " " and "\n" not in ws_pre):
-                    return LintResult(memory=memory, anchor=context.segment)
+                    result.append(
+                        LintResult(
+                            memory=memory,
+                            anchor=context.segment,
+                            description=f"Jinja tags should have a single "
```

Comment:
Updated the result to include the problem code, because there can be multiple Jinja tags in the same segment, but the position (line number and character position) will always point to the **start** of the segment, which could be confusing. (See the YAML test case -- it'll be clearer then.)